server/src/cloud/user.ts

```diff
@@ -0,0 +1,30 @@
+import {requireValidBoardMember} from "./permission";
+import {api} from "./util";
+
+/**
+ * @param showHiddenColumns Allows moderators to toggle the display of hidden columns (visible only to themselves)
+ */
+type EditUserConfigurationRequest = {
+  userId: string;
+  boardId: string;
+  editUserConfigurationRequest: {
+    showHiddenColumns: boolean;
+  };
+};
+
+export const initializeUserFunctions = () => {
+  /**
+   * Allows users to configure their settings.
+   */
+  api<EditUserConfigurationRequest, {status: string; description: string}>("editUserConfiguration", async (user, request) => {
+    await requireValidBoardMember(user, request.boardId);
+
+    if (request.editUserConfigurationRequest.showHiddenColumns !== undefined) {
```

Comment:
This setting should only be editable for moderators, shouldn't it?

Reply:
This is implicit done since only moderators have access to the `HeaderMenu`. But we can add an additional check only for this configuration to be sure.